File: agenkit-cs/src/Agenkit/Core/IntrospectionResult.cs
namespace Agenkit.Core;

/// <summary>
/// Represents a snapshot of an agent's internal state and capabilities.
/// </summary>
public record IntrospectionResult(
    string AgentName,
    IReadOnlyList<string> Capabilities,
    IReadOnlyDictionary<string, object>? Memory = null,
    IReadOnlyDictionary<string, object>? State = null,
    IReadOnlyList<string>? Tools = null);

File: agenkit-scala/src/main/scala/io/agenkit/core/IntrospectionResult.scala
package io.agenkit.core

/** Snapshot of an agent's runtime state for observability. */
case class IntrospectionResult(
  name: String,
  metadata: Map[String, Any] = Map.empty,
  capabilities: List[String] = List.empty,
  activeConnections: Int = 0,
  processedMessages: Long = 0L
)
